### PR TITLE
Fix and test for null value bug when using --repo_env

### PR DESCRIPTION
**Pull request by SalmaSamy**
Fix and test for null value bug when using --repo_env
[Issue Link](https://github.com/bazelbuild/bazel/issues/15430)

### CODE REVIEW

File: src/test/py/bazel/bazel_external_repository_test.py [hunk 1]
```diff
@@ -427,5 +427,34 @@ def testExternalBazelignoreContainingRepoName(self):
...
+      '    print("FOO", val)',
+      '    ctx.file("BUILD")',
+      'dump_env = repository_rule(',
+      '    implementation = _dump_env,',
+      '    local = True,',
+      ')'
+    ])
+
+    exit_code, _, stderr = self.RunBazel(
+      args=['build', '@debug//:all'],
```
Comment (meteorcloudy) [inline src/test/py/bazel/bazel_external_repository_test.py:452]: Please addd `-repo_env=FOO` here

Comment (SalmaSamy) [reply src/test/py/bazel/bazel_external_repository_test.py]: Done

**Review by meteorcloudy — APPROVED** [review]
Thanks for the fix!

**Review by meteorcloudy — APPROVED** [review]

**Comment by meteorcloudy** [comment]
@bazel-io fork 5.3.0

**Pull request closed without merging**